### DISCUSSION TITLE
Easy way to 'import' a NEURON model to Jaxley?

### DISCUSSION THREAD

**Discussion by briandepasquale** [jaxleyverse/jaxley#498] [General]
I'm quite new to jaxley, and it's been well over a decade since I used NEURON. There's a NEURON model from ModelDB (https://modeldb.science/118662) that I'd like to translate over to jaxley. Is there an easy/obvious way to do this, or to do it for some part of the model (e.g., just the morphology from `.hoc` files)? Thanks!

**Comment by michaeldeistler** [comment]
Hi there,

thanks for reaching out! Unfortunately, we do not yet support automated translation of NEURON models.

- for morphologies, Jaxley supports SWC files. Thus, you would have to export your morphology `.hoc` file (e.g. `dm1_morph_1.hoc`) to an `.swc` file. There are conversion tools available (e.g. [here](https://github.com/JustasB/hoc2swc)) but I have not tried them myself.
- for channels (or other mechanisms), @r-makarov built an automated tool to convert .mod files [here](https://dendrotweaks.readthedocs.io/en/latest/tutorials/convert_to_jaxley.html). Please post feedback [here](https://github.com/jaxleyverse/jaxley/discussions/668).


Please let us know if you need any more input or if anything is not working as expected!

All the best
Michael

**Reply by briandepasquale** [reply]
Thanks! I'll give the .hoc->.swc conversion tool a shot, and reach out with any challenges.

**Reply by michaeldeistler** [reply]
Cool! Keep us updated, I would be curious to hear if this works!

**Reply by briandepasquale** [reply]
Thanks! I gave it a shot and hit an error almost immediately. I opened an issue [issue](https://github.com/JustasB/hoc2swc/issues/2) with the developers but if you have any insight I'd welcome it!

**Reply by michaeldeistler** [reply]
Ah, too bad! I don't think I have any direct insights on why it's crashing unfortunately. Let's see what the devs say!

**Reply by kirkw-1** [reply]
Wow this would be a super useful tool, was there any progress on this?

"for channels (or other mechanisms), we are working on an automated tool that directly translates .nmodl files to Jaxley compatible channels"

**Reply by michaeldeistler** [reply]
Yes, there is an update! @r-makarov built such a tool for his DendroTweaks toolbox. See [here](https://dendrotweaks.readthedocs.io/en/latest/tutorials/convert_to_jaxley.html) for a tutorial on how to use it.

I have to admit that I have not tried it myself---any reports (both positive and negative) on how it works would be greatly appreciated! Please post feedback [here](https://github.com/jaxleyverse/jaxley/discussions/668).


Michael

**Comment by michaeldeistler** [comment]
Update:

@r-makarov built a tool to convert .mod files to Jaxley channels for his DendroTweaks toolbox. See [here](https://dendrotweaks.readthedocs.io/en/latest/tutorials/convert_to_jaxley.html) for a tutorial on how to use it.

Any reports (both positive and negative) on how it works would be greatly appreciated! Please post feedback [here](https://github.com/jaxleyverse/jaxley/discussions/668).

Michael
